src/main/java/uk/gov/hmcts/darts/retention/service/impl/ApplyRetentionCaseAssociatedObjectsSingleCaseProcessorImpl.java

```diff
@@ -73,14 +76,20 @@ public void processApplyRetentionToCaseAssociatedObjects(Integer caseId) {
     }
 
     private void applyRetentionToMedias(CourtCaseEntity courtCase) {
-
-        var medias = mediaRepository.findAllByCaseId(courtCase.getId());
-
-        for (var media : medias) {
-            var cases = media.associatedCourtCases();
-            if (allClosed(cases)) {
-                setLongestRetentionDateForMedia(media, cases);
-                setRetentionConfidenceScoreAndReasonForMedia(media, cases);
+        List<MediaEntity> mediaEntities = new ArrayList<>();
+        // get all current media linked to the case
+        mediaEntities.addAll(mediaRepository.findAllByCaseId(courtCase.getId()));
+        // get all media linked to the case via media linked case
+        mediaEntities.addAll(mediaRepository.findAllLinkedByMediaLinkedCaseAndCaseId(courtCase.getId()));
+        // remove duplicates
+        var allMedia = io.vavr.collection.List.ofAll(mediaEntities).distinctBy(MediaEntity::getId).toJavaList();
```

Comment:
Should have tests to verify only unique MediaEntity and CourtCaseEntities get processed